### DISCUSSION TITLE
Add a purchase Confirmation Button to the webstore

### DISCUSSION THREAD

**Discussion by Syroxxa** [gameserverapp/Platform#898] [Suggestions]
Hello there

It would be very nice if we could have a confirmation button after making a purchase on our Webstores. I got a lot of feedback from my community and often when they buy something on the webstore for ARK, they get it with 1 click (Order Now as shown on screenshot). This would just reduce the ammount of tickets to deal with refunding tokens to the playerbase. 

To make it simple:

After clicking order now, add an extra popup that asks the player to confirm his purchase.
![confirm](https://github.com/user-attachments/assets/ed8afa0d-06d3-46bc-a444-9528861b9e48)

**Comment by babsgosgens** [comment]
I agree with what you are trying to fix. I disagree with the method.

This just makes the ordering process unnecessarily more complex. Instead, I would propose to disable the button for the duration of the timer.
This will prevent re-ordering in a more user friendly way, while matching the underlying functionality (instead of trying to mask it).

I’m thinking a simple onclick handler could get the job done.

**Comment by Dovakillz** [comment]
Agreed

**Comment by mvaessen** [comment]
I'm not sure whether this will change anything. If people want to cancel, they can simply stop the process after clicking the button. The payment provider has `cancel` options too, which stops the whole process. GSA will not immediately charge someone after clicking that button. They need to click more buttons, before it's actually being charged. I don't see how adding an extra button will cure the actual problem.

**Reply by Syroxxa** [reply]
No this has nothing todo with the payment provider, just a website order for tokens instead.

**Reply by mvaessen** [reply]
Are you referring to shop packs or supporter tiers?

**Reply by Syroxxa** [reply]
Shop Packs sorry i should have said that before, not supporter tiers, they are absolutly fine.

**Reply by babsgosgens** [reply]
I believe @Syroxxa is referring to the cool-down timer when ordering packs. It can cause players to loose tokens on purchases that could not yet be delivered.

**Reply by Syroxxa** [reply]
@babsgosgens thats another point but not what im refering to,simply a purchase confirmation button after they order a shop pack for tokens :) Its nothing crazy but sometimes people from my community buy store packs accidently (idk how it can happen but i tell you it does) 🗡️

**Comment by mvaessen** [comment]
Okido, added to #922

**Reply by Syroxxa** [reply]
Thank you very much :)

**Comment by mvaessen** [comment]
![Screenshot_2024-10-08_at_12_51_46](https://github.com/user-attachments/assets/92cb240e-953c-4981-a350-c71827b711de)

This confirmation modal / popup will be added to all Shop pack `order now` buttons.

**Comment by babsgosgens** [comment]
let it be noted that I am still not in favour of this change. This makes sense on a checkout page with multiple items, not on a single item and for each item ordered.
